[Bug]: Hello, when I creating a datanode and a querynode container on docker host network, it gives an error which indicates port 9091 is in use.

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: 2.0.1
- Deployment mode(standalone or cluster):Cluster
- SDK version(e.g. pymilvus v2.0.0rc2):2.0.1
- OS(Ubuntu or CentOS): Ubuntu
- CPU/Memory: Enough
- GPU: None
- Others:
```


### Current Behavior

[2022/04/12 04:06:43.736 +00:00] [DEBUG] [base_table.go:65] ["config directory"] [configDir=/milvus/configs/]
[2022/04/12 04:06:43.737 +00:00] [ERROR] [metrics.go:653] ["handle metrics failed"] [error="listen tcp :9091: bind: address already in use"] [stack="github.com/milvus-io/milvus/internal/metrics.ServeHTTP.func1\n\t/go/src/github.com/milvus-io/milvus/internal/metrics/metrics.go:653"]

### Expected Behavior

All node and coord on same host network should be started without error message, make an environment variable to allow me allocate port for metrics rather than hardcoded to 9091.

### Steps To Reproduce

```markdown
1. Create docker host
2. Create Milvus datanode on Host network
3. Create Milvus querynode on Host network
```


### Anything else?

_No response_

--- a/internal/metrics/metrics.go
+++ b/internal/metrics/metrics.go
@@ -17,7 +17,11 @@
 package metrics
 
 import (
+	"fmt"
 	"net/http"
+	"os"
+	"strconv"
+
 	// nolint:gosec
 	_ "net/http/pprof"
 
@@ -28,6 +32,9 @@ import (
 )
 
 const (
+	DefaultListenPort = "9091"
+	ListenPortEnvKey  = "METRICS_PORT"
+
 	milvusNamespace = "milvus"
 
 	AbandonLabel = "abandon"
@@ -78,8 +85,20 @@ func ServeHTTP(r *prometheus.Registry) {
 	http.Handle("/metrics", promhttp.HandlerFor(r, promhttp.HandlerOpts{}))
 	http.Handle("/metrics_default", promhttp.Handler())
 	go func() {
-		if err := http.ListenAndServe(":9091", nil); err != nil {
+		bindAddr := getMetricsAddr()
+		log.Debug("metrics listen", zap.Any("addr", bindAddr))
+		if err := http.ListenAndServe(bindAddr, nil); err != nil {
 			log.Error("handle metrics failed", zap.Error(err))
 		}
 	}()
+}
+
+func getMetricsAddr() string {
+	port := os.Getenv(ListenPortEnvKey)
+	_, err := strconv.Atoi(port)
+	if err != nil {
+		return fmt.Sprintf(":%s", DefaultListenPort)
+	}
+
+	return fmt.Sprintf(":%s", port)
 }

--- a/internal/metrics/metrics_test.go
+++ b/internal/metrics/metrics_test.go
@@ -17,9 +17,11 @@
 package metrics
 
 import (
+	"os"
 	"testing"
 
 	"github.com/prometheus/client_golang/prometheus"
+	"github.com/stretchr/testify/assert"
 )
 
 func TestRegisterMetrics(t *testing.T) {
@@ -33,4 +35,12 @@ func TestRegisterMetrics(t *testing.T) {
 	RegisterProxy(r)
 	RegisterQueryNode(r)
 	RegisterQueryCoord(r)
+	ServeHTTP(r)
+}
+
+func TestGetMetricsAddr(t *testing.T) {
+	assert.Equal(t, getMetricsAddr(), ":"+DefaultListenPort)
+	testPort := "9092"
+	os.Setenv(ListenPortEnvKey, testPort)
+	assert.Equal(t, getMetricsAddr(), ":"+testPort)
 }

